cache.go

```diff
@@ -158,14 +161,18 @@ type Item struct {
 
 // NewCache returns a new Cache instance and any configuration errors, if any.
 func NewCache(config *Config) (*Cache, error) {
-	switch {
-	case config.NumCounters == 0:
-		return nil, errors.New("NumCounters can't be zero")
-	case config.MaxCost == 0:
+	if config.MaxCost == 0 {
 		return nil, errors.New("MaxCost can't be zero")
-	case config.BufferItems == 0:
-		return nil, errors.New("BufferItems can't be zero")
 	}
+
+	if config.NumCounters == 0 {
+		config.NumCounters = config.MaxCost * defaultNumCountersMult // sensible default
```

Comment:
Since MaxCost is the upper bound on size of cache in Bytes, it doesn't necessarily approximate to max number of keys we want in it. This likely leads to over-approximation for value of NumCounters which should be 10 times the number of keys in cache and not 10 times the size of cache.

Reply:
True, but I assumed over-approximation would be less harmful then under-approximation. Isn't this the case?